codegen-sbt/src/main/scala/caliban/codegen/CalibanSourceGenerator.scala

```diff
@@ -45,17 +45,19 @@ object CalibanSourceGenerator {
       interimPath.getParent.resolve(scalaName).toFile
   }
 
-  def collectSettingsFor(fileSettings: Seq[CalibanFileSettings], source: File): CalibanFileSettings = {
+  def collectSettingsFor(fileSettings: Seq[CalibanFileSettings], source: File): Seq[CalibanFileSettings] = {
     // Supply a default packageName.
     // If we do not, `src_managed.main.caliban-codegen-sbt` will be used,
     // which is not only terrible, but invalid.
     val defaults: CalibanCommonSettings = CalibanCommonSettings.empty.copy(packageName = Some("caliban"))
 
-    CalibanFileSettings(
-      file = source,
-      settings = fileSettings.collect { case needle if source.toPath.endsWith(needle.file.toPath) => needle }
-        .foldLeft[CalibanCommonSettings](defaults) { case (acc, next) => acc.combine(next.settings) }
-    )
+    val matchingSettingSets = fileSettings.collect {
+      case needle if source.toPath.endsWith(needle.file.toPath) => needle.settings
+    }.map(defaults.combine(_))
```

Comment:
This is the intended change in this PR, from what I can tell. I was `foldLeft`-ing in the beginning to permit tiered configuration, but that was before the `cs` pattern was discovered. I don't think it's useful to accumulate anymore, if the previous implementation was desired, it is possible to define a generic `CalibanSettings => CalibanSettings` and apply that before you chain the rest of your config.

Reply:
I'm not sure what that'd look like, can you provide some pseudocode?